pymc3/step_methods/hmc/base_hmc.py

```diff
@@ -171,12 +171,18 @@ def warnings(self):
         # Generate a global warning for divergences
         n_divs = self._num_divs_sample
         if n_divs and self._samples_after_tune == n_divs:
-            msg = ('The chain contains only diverging samples. The model is '
-                   'probably misspecified.')
+            message = ('The chain contains only diverging samples. The model '
+                       'is probably misspecified.')
             warning = SamplerWarning(
-                WarningType.DIVERGENCES, msg, 'error', None, None, None)
+                WarningType.DIVERGENCES, message, 'error', None, None, None)
+            warnings.append(warning)
+        elif n_divs == 1:
```

Comment:
There's a lot of repeated code here. Just use the `if` statement to change the message; not need to repeat the warning type code.

Reply:
That was my first thought as well, but if there are zero divergences, then we get an `E                   UnboundLocalError: local variable 'message' referenced before assignment`. See [any of the failed Travis builds here](https://travis-ci.org/pymc-devs/pymc3/builds/405556326?utm_source=github_status&utm_medium=notification) to see what I mean

Reply:
Sounds like a scoping issue. If you instantiate `message` outside the switch statement, it should work. So, something like this:

```python
        message = ''
        if n_divs and self._samples_after_tune == n_divs:
            message = ('The chain contains only diverging samples. The model '
                       'is probably misspecified.')
        elif n_divs == 1:
            message = ('There was 1 divergence after tuning. Increase '
                       '`target_accept` or reparameterize.')
        elif n_divs > 1:
            message = ('There were %s divergences after tuning. Increase '
                       '`target_accept` or reparameterize.'
                       % n_divs)
        if message:
            warning = SamplerWarning(
                WarningType.DIVERGENCES, message, 'error', None, None, None)
            warnings.append(warning)

```